pdns/dnsdistdist/dnsdist-tcp-downstream.cc

```diff
@@ -304,8 +304,35 @@ IOState TCPConnectionToBackend::sendQuery(std::shared_ptr<TCPConnectionToBackend
   return state;
 }
 
+class HandlingIOGuard
```

Comment:
I am not *really* a C++ programmer so I am not sure if all those operators are really needed; I just copy-pasted this from the other place this is used.